src/cli/cmd/iexec-app.js

```diff
@@ -1038,27 +1040,23 @@ run
 
       debug('requestorder', requestorder);
 
-      const totalCost = new BN(requestorder.appmaxprice)
-        .add(new BN(requestorder.datasetmaxprice))
-        .add(new BN(requestorder.workerpoolmaxprice))
-        .mul(new BN(requestorder.volume));
-
-      const { stake } = await checkBalance(chain.contracts, requester);
-      if (totalCost.gt(stake)) {
-        throw Error(
-          `Not enough RLC on your account (${formatRLC(
-            totalCost,
-          )} RLC required). Run "iexec account deposit" to topup your account.`,
-        );
-      }
+      const { total: totalCost, sponsored } = await estimateMatchOrders(
```

Comment:
Just noting that `estimateMatchOrders` will be performed twice:
 - Once here
 - And once again in the later called `matchOrders()`

Not ideal as some async work will be done twice, for nothing.

Would it be hard to change the `matchOrders()` signature and be able to give it `totalCost` and `sponsored` parameters?
Maybe in a next PR?

Reply:
yes, a refactoring of the matchOrders() function is definitely needed, but it won't be addressed in this PR